src/dwarf.rs

```diff
@@ -749,12 +750,15 @@ impl DwarfResolver {
     /// from the given file.  If the instance will be used for long
     /// running, you would want to load all data into memory to have
     /// the ability of handling all possible addresses.
-    pub fn open_for_addresses(
-        filename: &str,
+    pub fn open_for_addresses<P>(
+        filename: &P,
         addresses: &[u64],
         line_number_info: bool,
         debug_info_symbols: bool,
-    ) -> Result<DwarfResolver, Error> {
+    ) -> Result<DwarfResolver, Error>
+    where
+        P: AsRef<Path>,
```

Comment:
We probably need `AsRef<>` only at the public API exposed to users to make users' life easier.  For internal functions, they should get a reference of the type they want in the first place instead of delaying it until the last moment. I wonder if Rust will generate a separate version of the object code for each parameter type.

Reply:
The reason I did it is because it's the signature of [`File::open`](https://doc.rust-lang.org/std/fs/struct.File.html#method.open). In my opinion it makes sense to mirror it, unless we have a good reason not to. The user facing API is not the only one benefiting from a convenient to use API, internal ones will benefit as well.

> I wonder if Rust will generate a separate version of the object code for each parameter type.

Yes. For large functions we could introduce a non-generic inner one, as is a common pattern. I checked that all function are minimal in size or test-only.